_guides/developers/oauth2-token-generator.md

```diff
@@ -0,0 +1,98 @@
+---
+title: OAuth2 Token Generator
+excerpt: This page explains how to use the OAuth2 Token Generator to create tokens.
+summary: This page provides a comprehensive guide on using the OAuth2 
+  Token Generator tool for creating OAuth2 access and refresh tokens. 
+  It covers how to generate tokens with specific scopes and expiration 
+  times, and explains how these tokens can be used for API testing and development.
+permalink: /guides/developers/oauth2-token-generator/
+layout: integration
+topic: Developers
+jumbotron:
+  title: OAuth2 Token Generator
+  tagline: ""
+  breadcrumbs:
+  - label: Resources &raquo;
+    url: /resources/
+  - label: Guides &raquo;
+    url: /resources/guides/
+  - label: Developers &raquo;
+    url: /resources/guides/#developers
+---
+
+* TOC
+{:toc}
+
+# Introduction
+
+The OAuth2 Token Generator allows administrators to create access tokens for API authentication. These tokens provide secure access to Cerb's API with specific worker permissions and scopes. While OAuth2 typically requires implementing a complete authentication flow, the token generator provides a simpler way to create tokens directly.
+
+For detailed information about Cerb's OAuth2 authentication system, see the [API Authentication documentation](/docs/api/authentication/). If you're building custom integrations, you may also want to review:
+
+* [OAuth2 Token Validation example](/solutions/automations/validate-oauth2-token/) - Examples of validating tokens in webhook automations.
+
+* [Custom API guide](/guides/webhooks/custom-api/) - Build and authenticate APIs using webhook portals.
+
+# Prerequisites
+
+To use OAuth tokens you need to create an app in Cerb from **Search >> OAuth Apps >> (+)**.
+
+This will generate a random **Client ID** and **Client Secret** for you. The **Name:** should reflect your integration.
+
+If you want to allow Cerb workers to log in to your integration to generate tokens, the **Callback URL:** must be implemented by your app. If you're using the token generator in Cerb then the callback URL can be any value (e.g. your website).
+
+<div class="cerb-screenshot">
+<img src="/assets/images/guides/webhooks/custom-api/create-oauth-app.png" class="screenshot">
+</div>
+
+You can add your own scopes. The default provides an `api` scope that allows all endpoints and methods. Your automation will need to validate scopes.
+
+{% highlight yaml %}
+"api":
```

Comment:
It looks like the leading whitespace was munged on this one during paste. Happens sometimes. Try "Right-click -> Paste special -> Plain text" in phpStorm.